bears/rest/RSTcheckBear.py

```diff
@@ -32,16 +33,32 @@ class RSTcheckBear:
 
     @staticmethod
     def create_arguments(filename, file, config_file,
-                         code_block_language_ignore: list=()):
+                         code_block_language_ignore: typed_list(str)=(),
+                         directive_ignore: typed_list(str)=(),
+                         role_ignore: typed_list(str)=()):
         """
         :param code_block_language_ignore:
             Comma seperated list for languages of which code blocks
             should be ignored. Code block of following languages
             can be detected: ``bash``, ``c``, ``cpp``, ``json``,
             ``python``, ``rst``.
+
+        :param directive_ignore:
+             Comma separated list of directives to be ignored. Can be
+             used to ignore custom directives as well.
+
+        :param role_ignore:
+             Comma separated list of roles to be ignored. Can be used
+             to ignore custom roles.
         """
         args = ()
         if code_block_language_ignore:
             args = ('--ignore-language=' +
                     ','.join(code_block_language_ignore),)
-        return args + (filename,)
+        if directive_ignore:
+            directives = ','.join(part.strip() for part in directive_ignore)
+            args = ('--ignore-directives=' + directives,)
+        if role_ignore:
+            roles = ','.join(part.strip() for part in role_ignore)
+            args = ('--ignore-roles=' + roles,)
```

Comment:
also here, you overwrite `args` if one of the settings before was set :+1:

Reply:
oops